dn.go

```diff
@@ -143,6 +143,9 @@ func ParseDN(str string) (*DN, error) {
 			}
 		} else if char == ',' || char == '+' {
 			// We're done with this RDN or value, push it
+			if len(attribute.Type) == 0 {
+				return nil, errors.New("incomplete type, value pair")
```

Comment:
do we also need to check for `len(attribute.Type) == 0` in the `if char == '='` block, to catch `=foo,DC=example,DC=com`?